### PR TITLE
The slightly-good ninja nerf of 2024

### PR DESCRIPTION
**Pull request by Therealdoooc213**
## About The Pull Request

In the title. Nerfs ninjas, katana damage/bleed lowered (4 hits to crit instead of 3), suit protection values lowered, cant use guns anymore.

## Why It's Good For The Game

Ninja has been overpowered for some time, able to delimb in 3 hits, near infinite teleports, insane armor, and was able to use guns. This changes all of that.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.
image
image
image

</details>

## Changelog
:cl: Therealdoooc
balance: Ninjas can no longer use guns
balance: Ninja suit has lowered protection values
balance: The Energy Katana dashing ability takes longer to recharge now, has a lower damage, and deals lower bleeding
/:cl:

### CODE REVIEW

File: code/modules/antagonists/ninja/ninja.dm [hunk 1]
```diff
@@ -11,10 +11,12 @@
 /datum/antagonist/ninja/New()
 	if(helping_station)
 		can_elimination_hijack = ELIMINATION_PREVENT
-	. = ..()
+		. = ..()
```
Comment (EvilDragonfiend) [inline code/modules/antagonists/ninja/ninja.dm]: what's this indent

File: code/__DEFINES/traits/sources.dm [hunk 2]
```diff
@@ -133,3 +133,4 @@
 #define NO_GRAVITY_TRAIT "no-gravity"
 #define LIFECANDLE_TRAIT "lifecandle"
 #define LEAPER_BUBBLE_TRAIT "leaper-bubble"
+#define NINJA_TRAIT "ninja" //This is what happens to naughty little ninjas
```
Comment (EvilDragonfiend) [inline code/__DEFINES/traits/sources.dm]:
```suggestion
```
This is not trait.

File: code/modules/antagonists/ninja/ninja.dm [hunk 3]
```diff
@@ -11,10 +11,12 @@
...
+
 
 /datum/antagonist/ninja/apply_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
+	ADD_TRAIT(M, TRAIT_NOGUNS, NINJA_TRAIT)
```
Comment (EvilDragonfiend) [inline code/modules/antagonists/ninja/ninja.dm]:
```suggestion
	ADD_TRAIT(M, TRAIT_NOGUNS, FAST_REF(src))
```
But also do `Remove Trait()` from antag removal proc.

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
Incorrect practice of trait system. NINJA_TRAIT is not how its name means, and trait source shouldn't be given by such method.

**Comment by Rukofamicom** [comment]
The ninja, like wizards, are supposed to be an exceedingly powerful enemy to fight.

**Comment by PowerfulBacon** [comment]
This PR is a nerf for the sake of nerfing and fails to understand what is making the ninja so powerful:
- ridiculous speed
- overpowered katana
- impossible to chase due to overpowered movement options
- invisibility that grants laser immunity usable in combat

If we nerf them in this way, we are making them harder rather than simply less powerful making now player Ninjas worse while not addressing the issues that make skilled players too powerful with ninja.

**Comment by PowerfulBacon** [comment]
Now that the other ninja nerf is in, I think it's best that we look at the effects of that and reconsider this going forward

**Pull request closed without merging**